src/RequestExecutor/WsRequestExecutor.spec.ts

```diff
@@ -0,0 +1,179 @@
+        { lifecycle: Lifecycle.Singleton }
+      );
+  });
+
+  afterEach(() => {
+    container.reset();
+
+    reset(requestMock);
+  });
+
+  describe('protocol', () => {
+    it('should use WS protocol', () => {
+      const executor = container.resolve<RequestExecutor>(RequestExecutor);
+
+      executor.protocol.should.equal(Protocol.WS);
+    });
+  });
+
+  describe('execute', () => {
+    let server: Server;
+    let wsPort: number;
+
+    beforeEach((done) => {
+      server = new Server({ port: 0 });
+      server.on('listening', () => {
+        const address = server.address();
+        if (typeof address === 'string') {
+          return done(new Error('Unsupported server address type'));
+        }
+
+        wsPort = address.port;
+
+        done();
+      });
+    });
+
+    afterEach((done) => {
+      wsPort = Number.NaN;
+      server.close(done);
+    });
+
+    it('should call setCerts on the provided request if there were certificates configured globally', async () => {
+      when(requestMock.url).thenReturn('wss://foo.bar');
+      when(requestMock.headers).thenReturn({});
+      const request = instance(requestMock);
+      requestExecutorOptions.certs = [];
+      const executor = container.resolve<RequestExecutor>(RequestExecutor);
+
+      await executor.execute(request);
+
+      verify(requestMock.setCerts(anything())).once();
+    });
+
+    it('should not call setCerts on the provided request if there were no globally configured certificates', async () => {
+      when(requestMock.url).thenReturn('wss://foo.bar');
+      when(requestMock.headers).thenReturn({});
+      const request = instance(requestMock);
+      const executor = container.resolve<RequestExecutor>(RequestExecutor);
+
+      await executor.execute(request);
+
+      verify(requestMock.setCerts(anything())).never();
+    });
+
+    it('should send request body to a web-socket server', (done) => {
+      const url = `ws://localhost:${wsPort}`;
+      const headers = {};
+      const body = 'test request body';
+      const request = new Request({ url, headers, body });
+
+      server.on('connection', async (socket) => {
+        socket.on('message', (data) => {
+          data.should.be.instanceOf(Buffer);
+          data.toString().should.equal(body);
+
+          socket.send('test reply');
+
+          done();
+        });
+      });
+
+      const executor = container.resolve<RequestExecutor>(RequestExecutor);
+
+      executor.execute(request);
+    });
+
+    it('should fail sending request by timeout', async () => {
+      requestExecutorOptions.timeout = 1;
+
+      const url = `ws://localhost:${wsPort}`;
+      const request = new Request({ url, headers: {} });
+      const executor = container.resolve<RequestExecutor>(RequestExecutor);
+
+      const response = await executor.execute(request);
+
+      response.message.should.equal('Waiting frame has timed out');
+      response.errorCode.should.equal('ETIMEDOUT');
```

Comment:
Assert on the entire object, not on every field. It's important from the test report perspective since we want to desire a result as a single unit, like a story. To be honest, it is not so convenient while using chai.

```suggestion
      response.should.dep.equal({ errorCode: 'ETIMEDOUT', message: 'Waiting frame has timed out' });
```